src/models/dependency-injection/provider.service.ts

```diff
@@ -60,11 +51,22 @@ class ProviderService {
       interceptor: this.providerMaps.interceptor.get(provider),
       transformer: this.providerMaps.transformer.get(provider),
       middleware: this.providerMaps.middleware.get(provider),
+      microservice: this.providerMaps.microservice.get(provider),
     };
   }
 
-  getAllInstances() {
-    return this.providerMaps;
+  getProviderByType(type: ProviderType, provider: string): ProviderInterface {
+    return this.providerArray.filter((prov) => prov.type === type && prov.name === provider)[0];
```

Comment:
Could use .find(). Also, when this function is called should we assume the data is always found?
```suggestion
    return this.providerArray.find((prov) => prov.type === type && prov.name === provider);
```